Realm/RealmSwift/Realm.swift

```diff
@@ -21,27 +21,135 @@ import Realm.Private
+
+:returns: Results with all objects of the given type in the specified Realm.
+*/
+public func objects<T: Object>(type: T.Type, inRealm realm: Realm = defaultRealm()) -> Results<T> {
+    return Results<T>(RLMGetObjects(realm.rlmRealm, T.className(), nil))
 }
 
 // MARK: Default Realm Helpers
 
-public func defaultRealmPath() -> String {
-    return RLMRealm.defaultRealmPath()
+/**
+The location of the default Realm as a string. Can be overridden.
+
+`~/Application Support/{bundle ID}/default.realm` on OS X.
+
+`default.realm` in your application's documents directory on iOS.
+
+:returns: Location of the default Realm.
+*/
+public var defaultRealmPath: String {
+    set {
+        RLMRealm.setDefaultRealmPath(newValue)
+    }
+    get {
+        return RLMRealm.defaultRealmPath()
+    }
 }
 
+/**
+Obtains an instance of the default Realm.
+
+The default Realm is persisted as default.realm under the Documents directory of
+your Application on iOS, and in your application's Application Support
+directory on OS X.
+
+:returns: The default `Realm` instance for the current thread.
+*/
 public func defaultRealm() -> Realm {
-    return Realm(rlmRealm: RLMRealm.defaultRealm())
+    return Realm(RLMRealm.defaultRealm())
+}
+
+/**
+Set the encryption key to use when opening Realms at a certain path.
+
+This can be used as an alternative to explicitly passing the key to
+`Realm(path:, encryptionKey:, readOnly:, error:)` each time a Realm instance is
+needed. The encryption key will be used any time a Realm is opened with
+`Realm(path:)` or `defaultRealm()`.
+
+If you do not want Realm to hold on to your encryption keys any longer than
+needed, then use `Realm(path:, encryptionKey:, readOnly:, error:)` rather than this
+method.
+
+:param: encryptionKey 64-byte encryption key to use, or `nil` to unset.
+:param: path          Realm path to set the encryption key for.
+*/
+public func setEncryptionKey(encryptionKey: NSData?, forRealmsAtPath path: String) {
+    RLMRealm.setEncryptionKey(encryptionKey, forRealmsAtPath: path)
 }
 
-public class Realm {
+/**
+A Realm instance (also referred to as "a realm") represents a Realm
+database.
+
+Realms can either be stored on disk (see `init(path:)`) or in
+memory (see `init(inMemoryIdentifier:)`).
+
+Realm instances are cached internally, and constructing equivalent Realm
+objects (with the same path or identifier) multiple times on a single thread
+within a single iteration of the run loop will normally return the same
+Realm object. If you specifically want to ensure a Realm object is
+destroyed (for example, if you wish to open a realm, check some property, and
+then possibly delete the realm file and re-open it), place the code which uses
+the realm within an `autoreleasepool {}` and ensure you have no other
+strong references to it.
+
+:warning: Realm instances are not thread safe and can not be shared across
+          threads or dispatch queues. You must call this method on each thread you want
```

Comment:
I'll change here and in `RLMRealm` to "you must construct a new instance on each thread..."